src/main/java/seedu/address/model/volunteer/Volunteer.java

```diff
@@ -0,0 +1,175 @@
+package seedu.address.model.volunteer;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Volunteer in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Volunteer {
+    // Used to keep track of current max Id in the system
+    private static int maxId = 0;
+
+    // Identity fields
+    private final VolunteerId volunteerId;
+    private final Name name;
```

Comment:
Remember to include the following fields for volunteers!
* Birthday
* Gender

Reply:
Added Gender and Birthday classes, but have yet to implement the correct methods.